src/ui-event.c

```diff
@@ -352,3 +352,28 @@ bool char_matches_key(wchar_t c, keycode_t key)
 	text_mbstowcs(keychar, k, 1);
 	return (c == keychar[0]);
 }
+
+/**
+ * Check if a UI event matches a certain keycode ('a', 'b', etc)
+ */
+bool event_is_key(ui_event e, keycode_t key)
```

Comment:
I couldn't really find any convenient ways to check input for specific key/mouse events so I added some. Did I miss ones that already exist?

Other potential names for functions like this:
* `ui_event_is_key` (more accurate but longer)
* `input_is_key` (if event seems too generic)
* `evt_is_key` (very short)

Reply:
I don't think you missed anything.  I think the only possible ambiguity is with game events, and the context makes it very clear, so this name is fine.